ocaml/xapi/xapi_event.ml

```diff
@@ -693,6 +701,14 @@ let from_inner __context session subs from from_t deadline =
   {events; valid_ref_counts; token= Token.to_string (last, msg_gen)}
 
 let from ~__context ~classes ~token ~timeout =
+  let deadline = Unix.gettimeofday () +. timeout in
```

Comment:
They can be used, this usecase is the main reason I created timers: https://github.com/xapi-project/xen-api/blob/master/ocaml/libs/clock/timer.mli

Reply:
This code was written months before the Clock module existed.
You are right that I could change it to better take advantage of what exists in XAPI currently.

I don't want to change too much in a single PR though, because using Clock here is unrelated to what I'm trying to fix with batching.
I'll probably send a separate PR to change the existing Xapi_event module to use the Clock module first, and then rebase this PR on top of it.

Reply:
I'll put this PR back into draft until I prepare the other one.

Reply:
Rebased on top of the mtime PR.